Add tutorial to use debugging in vscode for unit tests (#3623)
realm/realm-js: contrib/vscode-debugging.md

# VSCode Debugging

## Setup

First make sure your environment is setup by following the [building instructions](./building.md)

Then you will need the following plugins for VSCode:
* [C/C++ by Microsoft](https://github.com/Microsoft/vscode-cpptools)
* [CodeLLDB](https://github.com/vadimcn/vscode-lldb)

## Building

In order to debug C++, you must first build the project in debug mode.  This can be done with the command

```bash

> $ npm run rebuild --debug

Reviewer: This won't work on v11+ (`develop`), will it?  We should state that here or provide alternative instructions.

Author: I did test this in develop actually.

```

After this is completed there should be a `debug` folder visible in your `compiled` directory.  If not, perhaps delete the `compiled` directory and try again.

## Debugging Realm Unit Tests through Example

First lets take a look at `.vscode/launch.json`.  This contains various ways to launch commands from vscode.  For this example we will look at the command `LLDB Launch Unit Tests`.

```json
{
    "type": "lldb",
    "request": "launch",
    "name": "LLDB Launch Unit Tests",
    "program": "node",
    "args": [
        "--expose_gc",

>         "${workspaceRoot}/tests/node_modules/jasmine/bin/jasmine.js",

Reviewer: ```suggestion
        "${workspaceFolder}/tests/node_modules/jasmine/bin/jasmine.js",
```

        "spec/unit_tests.js",
        "--filter=."
    ],
    "cwd": "${workspaceFolder}/tests"
}
```

> A quick read through this code shows that the launch type is `lldb` and it is using the `node` command to invoke the `jasmine` test framework with our `spec/unit_tests.js`.  The filter option is currently running all tests.  To make things easier, lets modify that param to take a single test (`testListPush` from `tests/list-tests.js`).

Reviewer: ```suggestion
A quick read through this code shows that the launch type is `lldb` provided by the CodeLLDB extension, and it is using the `node` command to invoke the `jasmine` test framework with our `spec/unit_tests.js`.  The filter option is currently running all tests.  To make things easier, let's modify that parameter to take a single test (`testListPush` from `tests/list-tests.js`).
```

```json
{
    "type": "lldb",
    "request": "launch",
    "name": "LLDB Launch Unit Tests",
    "program": "node",
    "args": [
        "--expose_gc",

>         "${workspaceRoot}/tests/node_modules/jasmine/bin/jasmine.js",

Reviewer: ```suggestion
        "${workspaceFolder}/tests/node_modules/jasmine/bin/jasmine.js",
```
